legend-depot-store-api/src/main/java/org/finos/legend/depot/store/admin/api/metrics/QueryMetricsStore.java

```diff
@@ -30,4 +31,5 @@
 
     long consolidate(VersionQueryMetric metric);
 
+    List<VersionQueryMetric> findLeastRecentlyUsed(String regexPattern, Date date);
```

Comment:
this is not really doing least recently used ....is a filter by date.. findMetricsBefore ,. would be a better name
also can you think of a way to exclude esnapsho the regex is not areally good way to spwecify the search..
whats the method doing...finding version metrics before date or finding snapshot metrics before  a date